kmir/src/kmir/smir.py

```diff
@@ -329,8 +329,9 @@ def metadata_from_json(typeinfo: dict) -> TypeMetadata:
 
 def _decode(bytes: list[int]) -> int:
     # assume little-endian: reverse the bytes
-    bytes.reverse()
-    return reduce(lambda x, y: x * 256 + y, bytes)
+    bs = bytes.copy()
+    bs.reverse()
+    return reduce(lambda x, y: x * 256 + y, bs)
```

Comment:
This one was fun... `bytes.reverse()` modified the SMIR JSON data for cases where `_decode` was called...